Realm/RLMRealm.mm

```diff
@@ -706,6 +706,83 @@ - (void)cancelWriteTransaction {
     }
 }
 
+#ifdef REALM_ASYNC_WRITES
+
+- (BOOL)inAsyncWriteTransaction {
+    return _realm->is_in_async_transaction();
+}
+
+- (RLMAsyncTransactionId)beginAsyncWriteTransaction:(void(^)())block {
+    try {
+        _realm->async_begin_transaction(block);
+    }
+    catch (std::exception &ex) {
+        @throw RLMException(ex);
+    }
+}
+
+- (RLMAsyncTransactionId)commitAsyncWriteTransaction {
```

Comment:
This should probably just return void? You can't do anything with the returned transaction id as there's no completion callback to cancel.